content/en/blog/_posts/clientcmd-tutorial.md

```diff
@@ -0,0 +1,292 @@
+If a setting is not defined in a map, the last definition wins.
+
+When settings are retrieved using `KUBECONFIG`,
+missing files result in warnings only.
+If the user explicitly specifies a path (in `--kubeconfig` style),
+there must be a corresponding file.
+
+If `KUBECONFIG` isn't defined,
+the default configuration file, `~/.kube/config`, is used instead,
+if present.
+
+### Overall process
+
+The general usage pattern is succinctly expressed in
+[the package documentation](https://pkg.go.dev/k8s.io/client-go/tools/clientcmd):
+
+```go
+loadingRules := clientcmd.NewDefaultClientConfigLoadingRules()
+// if you want to change the loading rules (which files in which order), you can do so here
+
+configOverrides := &clientcmd.ConfigOverrides{}
+// if you want to change override values or bind them to flags, there are methods to help you
+
+kubeConfig := clientcmd.NewNonInteractiveDeferredLoadingClientConfig(loadingRules, configOverrides)
+config, err := kubeConfig.ClientConfig()
+if err != nil {
+	// Do something
+}
+client, err := metav1.New(config)
+// ...
+```
+
+In the context of this article, there are six steps:
+
+1. [Configure the loading rules](#configure-the-loading-rules).
+1. [Configure the overrides](#configure-the-overrides).
+1. [Build a set of flags](#build-a-set-of-flags).
+1. [Bind the flags](#bind-the-flags).
+1. [Build the merged configuration](#build-the-merged-configuration).
+1. [Obtain an API client](#obtain-an-api-client).
+
+### Configure the loading rules
+
+`clientcmd.NewDefaultClientConfigLoadingRules()` builds loading rules which will use either the contents of the `KUBECONFIG` environment variable,
+or the default configuration file name (`~/.kube/config`).
+In addition, if the default configuration file is used,
+it is able to migrate settings from the (very) old default configuration file
+(`~/.kube/.kubeconfig`).
+
+You can build your own `ClientConfigLoadingRules`,
+but in most cases the defaults are fine.
+
+### Configure the overrides
+
+`clientcmd.ConfigOverrides` is a `struct` storing overrides which will be applied over the settings loaded from the configuration derived using the loading rules.
+In the context of this article,
+its primary purpose is to store values obtained from command line arguments.
+These are handled using the [pflag](https://github.com/spf13/pflag) library,
+which is a drop-in replacement for Go's [flag package](https://pkg.go.dev/flag),
+adding support for double-hyphen arguments with long names.
+
+In most cases there's nothing to set in the overrides;
+we will only bind them to flags.
+
+### Build a set of flags
+
+In this context, a flag is a representation of a command line argument,
+specifying its long name (such as `--namespace`),
+its short name if any (such as `-n`),
+its default value,
+and a description shown in the usage information.
+Flags are stored in instances of
+[the `FlagInfo` `struct`](https://pkg.go.dev/k8s.io/client-go/tools/clientcmd#FlagInfo).
```

Comment:
```suggestion
the [`FlagInfo`](https://pkg.go.dev/k8s.io/client-go/tools/clientcmd#FlagInfo) struct.
```

We prefer using the title or name of the thing we're linking to as the hyperlink text.